src/model/utils/selection-post-fixer.js

```diff
@@ -263,6 +249,35 @@ function checkSelectionOnNonLimitElements( start, end, schema ) {
 	return startIsOnBlock || endIsOnBlock;
 }
 
+// Returns a minimal non-intersecting array of ranges.
+//
+// @param {Array.<module:engine/model/range~Range>} ranges
+// @returns {Array.<module:engine/model/range~Range>}
+function mergeIntersectingRanges( ranges ) {
```

Comment:
Needs a fix for adjacent ranges (valid ones).

Reply:
That must have been some false negative :relieved: - probably due to refactoring in table selection PR and the way manual tests are built.

The two tests that I've added are for the case that "stoped" working for me during the aforementioned refactoring.

It is OK after all :)